src/S3/S3Client.php

```diff
@@ -499,6 +527,66 @@ public static function _apply_use_arn_region($value, array &$args, HandlerList $
         }
     }
 
+    public static function _default_request_checksum_calculation(array $args): string
+    {
+        return ConfigurationResolver::resolve(
+            'request_checksum_calculation',
+            ApplyChecksumMiddleware::DEFAULT_CALCULATION_MODE,
+            'string',
+            $args
+        );
+    }
+
+    public static function _apply_request_checksum_calculation(
+        string $value,
+        array &$args
+    ): void
+    {
+        $value = strtolower($value);
```

Comment:
What happens if a customer provide an erroneous value for the parameter  `request_checksum_calculation` being resolved here?. Will it just ignored?

Reply:
I should have thrown an exception there and for the other example you pointed out. Will update.